saleor/graphql/product/mutations/products.py

```diff
@@ -688,14 +689,16 @@ def perform_mutation(cls, _root, info, **data):
         instance = cls.get_node_or_error(info, node_id, only_type=Product)
         variants_id = list(instance.variants.all().values_list("id", flat=True))
         # get draft order lines for variant
-        line_pks = list(
-            order_models.OrderLine.objects.filter(
-                variant_id__in=variants_id, order__status=OrderStatus.DRAFT
-            ).values_list("pk", flat=True)
+        lines = order_models.OrderLine.objects.select_related("order").filter(
+            variant__pk__id=variants_id, order__status=OrderStatus.DRAFT
         )
+        line_pks = [line.pk for line in lines]
+        orders = {line.order for line in lines}
         response = super().perform_mutation(_root, info, **data)
         # delete order lines for deleted variant
         order_models.OrderLine.objects.filter(pk__in=line_pks).delete()
+        for order in orders:
+            recalculate_order_prices(order, save=True)
```

Comment:
IMO, you should call `recalculate_order` which recalculates the prices and also the weight of the order (which is also different now).